mne_connectivity/decoding/decomposition.py

```diff
@@ -421,9 +417,13 @@ def transform(self, X):
                 "no filters are available, please call the `fit` method first"
             )
 
-        # transform seed and target data
-        X_seeds = self.filters_[0].T @ X[..., self.indices[0], :]
-        X_targets = self.filters_[1].T @ X[..., self.indices[1], :]
+        # transform seed and target data (i=channels; j=components; k=epochs; l=times)
+        if X.ndim == 2:
+            dim_subs = "ij,il->jl"
+        else:
+            dim_subs = "ij,kil->kjl"
+        X_seeds = np.einsum(dim_subs, self.filters_[0], X[..., self.indices[0], :])
+        X_targets = np.einsum(dim_subs, self.filters_[1], X[..., self.indices[1], :])
```

Comment:
I don't think you need any `einsum` here and can just make use of the fact that `@` / `matmul` broadcasts, i.e. taking `i=11, j=12, k=13, l=14`:
```
>>> (np.zeros((11, 12)).T @ np.zeros((13, 11, 14))).shape
(13, 12, 14)
```
so you can put back what you had before
```suggestion
        X_seeds = self.filters_[0].T @ X[..., self.indices[0], :]
        X_targets = self.filters_[1].T @ X[..., self.indices[1], :]
```

Reply:
Mb, changed to this simpler approach. Thanks!